src/test/java/net/lingala/zip4j/crypto/PBKDF2/BinToolsTest.java

```diff
@@ -0,0 +1,52 @@
+package net.lingala.zip4j.crypto.PBKDF2;
+
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class BinToolsTest {
+
+  @Rule
+  public final ExpectedException thrown = ExpectedException.none();
+
+  @Test
+  public void bin2hexForValidInputReturnsValidHex() {
+    final byte[] b = {(byte) 112};
+    assertThat(BinTools.bin2hex(b)).isEqualTo("70");
+  }
+
+  @Test
+  public void bin2hexForEmptyInputReturnsEmptyString() {
+    assertThat(BinTools.bin2hex(new byte[]{})).isEqualTo("");
+  }
+
+  @Test
+  public void bin2hexForNullInputReturnsEmptyString() {
+    assertThat(BinTools.bin2hex(null)).isEqualTo("");
+  }
+
+  @Test
+  public void bin2hexForNullInputReturnsEmptyArray() {
+    final String s = null;
+    assertThat(BinTools.hex2bin(s)).isEqualTo(new byte[]{});
```

Comment:
Unnecessary variable for a null. You can directly pass null instead of `s` in the assert.